src/coresyftools/coresyftools/tool.py

```diff
@@ -38,9 +59,27 @@ def __init__(self, run_script_file_name):
             self.context_directory, self.MANIFEST_FILE_NAME)
         self.manifest = get_manifest(self.manifest_file_name)
         self.arg_parser = CoReSyFArgumentParser(self.manifest)
+        if 'command' in self.manifest:
+            self._validate_command(self.manifest['command'])
         self.operation = self.manifest.get('operation', {})
         self._validate_operation(self.operation)
 
+    def _validate_command(self, command):
+        placeholders = self._extract_command_placeholders(command)
+        for placeholder in placeholders:
```

Comment:
If we have two methods: `has_unexpected_placeholders` and `has_missing_placeholders` wouldn't this be more readable and isolated?
(Opinion based review)

Reply:
Maybe, but that way we would lose the information of what placeholder was unexpected or missing. Furthermore, the logics command validation could easily change in a way that different (or none) exception case are distinguished.

Reply:
If you'd extract the `if`, you would still know the placeholder you're referring to. You would simply pass the `placeholder` as a parameter argument. Either way, I leave the decision to you.